lib.nix

```diff
@@ -10,133 +10,152 @@
           };
-          }
-          ```
-
-          Example output:
-          ```
-          {
-          "a.b.c" = <path>;
-          }
-          ```
-          *
-          */
-          tree: let
-            op = sum: path: val: let
-              pathStr = builtins.concatStringsSep "." path; # dot-based reverse DNS notation
-            in
-              if builtins.isPath val
-              then
-                # builtins.trace "${toString val} is a path"
-                (sum
-                  // {
-                    "${pathStr}" = val;
-                  })
-              else if builtins.isAttrs val
-              then
-                # builtins.trace "${builtins.toJSON val} is an attrset"
-                # recurse into that attribute set
-                (recurse sum path val)
-              else
-                # ignore that value
-                # builtins.trace "${toString path} is something else"
-                sum;
-
-            recurse = sum: path: val:
-              builtins.foldl'
-              (sum: key: op sum (path ++ [key]) val.${key})
-              sum
-              (builtins.attrNames val);
-          in
-            recurse {} [] tree;
-
-        rakeLeaves =
-          /*
-          *
-          Synopsis: rakeLeaves _path_
-
-          Recursively collect the nix files of _path_ into attrs.
-
-          Output Format:
-          An attribute set where all `.nix` files and directories with `default.nix` in them
-          are mapped to keys that are either the file with .nix stripped or the folder name.
-          All other directories are recursed further into nested attribute sets with the same format.
-
-          Example file structure:
-          ```
-          ./core/default.nix
-          ./base.nix
-          ./main/dev.nix
-          ./main/os/default.nix
-          ```
-
-          Example output:
-          ```
-          {
-          core = ./core;
-          base = base.nix;
-          main = {
+        };
+      }
+      => { "a.b.c" = <path>; }
+      ```
+      */
+      flattenTree = tree: let
+        op = sum: path: val: let
+          pathStr = builtins.concatStringsSep "." path; # dot-based reverse DNS notation
+        in
+          if builtins.isPath val
+          then
+            # builtins.trace "${toString val} is a path"
+            (sum
+              // {
+                "${pathStr}" = val;
+              })
+          else if builtins.isAttrs val
+          then
+            # builtins.trace "${builtins.toJSON val} is an attrset"
+            # recurse into that attribute set
+            (recurse sum path val)
+          else
+            # ignore that value
+            # builtins.trace "${toString path} is something else"
+            sum;
+
+        recurse = sum: path: val:
+          builtins.foldl'
+          (sum: key: op sum (path ++ [key]) val.${key})
+          sum
+          (builtins.attrNames val);
+      in
+        recurse {} [] tree;
+
+      /*
```

Comment:
The nix code formatter destroy doc comments just now. But this is a different pull request to fix it.